scripts/console.ts

```diff
@@ -283,29 +303,43 @@ export async function createClones(
     return [tokenAddr, governorAddr];
 }
 
-// --- Deployment and input handling helpers ---
+// --- Deployment helpers ---
 
-// source of most of these are from Radicle:
-// https://github.com/radicle-dev/radicle-contracts/blob/7070d51fdd8f99790b8fb4e4c953351fd417839a/src/deploy-to-network.ts
+const JSON_NUM_SPACES = 4;
 
-async function deploy<T extends Contract>(name: string, fn: () => Promise<T>): Promise<T> {
+// eslint-disable-next-line @typescript-eslint/no-var-requires, node/no-unpublished-require
+let deployments: Deployments = require(`../deployments.json`);
+
+async function trackDeployment<T extends Contract>(
+    fn: () => Promise<T>,
+    name: string = `Contract`,
+): Promise<T> {
     for (;;) {
         try {
-            console.log(`Deploying`, name, `contract...`);
+            console.log(`Deploying ${name} ...`);
 
             const contract = await fn();
             const net = await contract.provider.getNetwork();
 
-            console.log(name, `address:`, etherscanAddress(net.name, contract.address));
+            console.log(`Deployer address: ${contract.deployTransaction.from}`);
+            console.log(`${name} address: ${etherscanAddress(net.name, contract.address)}`);
             console.log(
-                name,
-                `transaction:`,
-                etherscanTx(net.name, contract.deployTransaction.hash),
+                `${name} transaction: ${etherscanTx(net.name, contract.deployTransaction.hash)}`,
             );
-            if (contract.deployTransaction.gasPrice) {
-                console.log(`Gas price:`, contract.deployTransaction.gasPrice.toString(), `wei`);
+
+            if (
+                contract.deployTransaction.blockNumber &&
+                contract.deployTransaction.gasPrice &&
+                contract.deployTransaction.gasLimit
+            ) {
+                console.log(`Block number: ${contract.deployTransaction.blockNumber.toString()}`);
```

Comment:
If we end up creating a `CONFIG_GAS` environment variable, then I think it'd make sense to wrap the block number logging statement under `VERBOSE`.

Reply:
Added, checks `GAS_MODE` such that we only get the bottom 3 lines when true:

```txt
Enter the contract to deploy [1, 2, 3, 0]: 2
Deploying TokenFactory ...
TokenImpl address: 0xd515F10641B571F2E008285e499060f6Df8a02DC
TokenFactory address: 0x6F2A1d800c9B7A6b4C1Ebc2f45F33885C71C7388
Deployer address: 0x55D935Ab6fE0F9a7fD2C7AB7635e164d3eD1AD24
TokenFactory address: https://goerli.arbiscan.io/address/0x6F2A1d800c9B7A6b4C1Ebc2f45F33885C71C7388
TokenFactory transaction: https://goerli.arbiscan.io/tx/0x8f5eaa60f4494f7a7f4967681aea8f8cd338bda885c3f780c5c5bd09ebab90ee
Block number: 909328
Gas price: 0.1 Gwei
Gas limit: 6086160 Wei
```

This will definitely be a more relevant once we start doing transactions too, as people are less likely to look at linked scanner for each one but the additional reporting may be wanted.